File: frontends/storefront-next/src/features/tenant/types/tenant.ts
export type TenantTheme = {
  primaryColor: string;
  secondaryColor: string;
  accentColor: string;
  backgroundColor: string;
  surfaceColor: string;
  textColor: string;
  mutedTextColor: string;
  headingFont: string;
  bodyFont: string;
  borderRadius: string;
  announcementEnabled: boolean;
  announcementText: string | null;
  announcementUrl: string | null;
};

export type StorefrontSettings = {
  isPublished: boolean;
  contactFormEnabled: boolean;
  showPrices: boolean;
  defaultCatalogSort: "FEATURED" | "NEWEST" | "PRICE_ASC" | "PRICE_DESC" | "NAME_ASC";
  catalogColumnsDesktop: 2 | 3 | 4;
};

export type StoreContactChannel = {
  type: "WHATSAPP" | "FACEBOOK" | "INSTAGRAM" | "OTHER";
  value: string;
  url: string;
  enabled: boolean;
  sortOrder: number;
};

export type TenantStorefrontContent = {
  account?: {
    enabled: boolean;
  };
  hero: {
    eyebrow: string;
    title: string;
    description: string;
    imageUrl: string;
    imageAlt: string;
  };
  benefits: Array<{
    icon: "care" | "pickup" | "financing" | "security";
    title: string;
    description: string;
  }>;
  whatsapp: {
    title: string;
    description: string;
    message: string;
  };
  footerDescription: string;
};

export type TenantConfig = {
  id: string;
  slug: string;
  name: string;
  shortName: string;
  logo?: string;
  defaultCurrency: string;
  timeZone: string;
  contactEmail: string | null;
  phone: string | null;
  addressLine: string | null;
  addressNumber: string | null;
  city: string | null;
  province: string | null;
  postalCode: string | null;
  countryCode: string;
  enabled: boolean;
  settings: StorefrontSettings;
  theme: TenantTheme;
  contactChannels: StoreContactChannel[];
  storefront: TenantStorefrontContent;
};

File: backends/dotnet/Siuden.Domain/Entities/Tenant.cs
using Siuden.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Siuden.Domain.Entities;

public class Tenant : AuditableEntity<Guid>
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string BrandName { get; set; } = string.Empty;
    public string ContactEmail { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string AddressLine { get; set; } = string.Empty;
    public string AddressNumber { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string Province { get; set; } = string.Empty;
    public string PostalCode { get; set; } = string.Empty;
    public string CountryCode { get; set; } = string.Empty;
    public string PrimaryColor { get; set; } = string.Empty;
    public string SecondaryColor { get; set; } = string.Empty;
    public string BackgroundColor { get; set; } = string.Empty;
    public string TextColor { get; set; } = string.Empty;
    public string HeadingFont { get; set; } = string.Empty;
    public string BodyFont { get; set; } = string.Empty;
    public string BorderRadius { get; set; } = string.Empty;
    public bool AnnouncementEnabled { get; set; }
    public string AnnouncementText { get; set; } = string.Empty;
    public string? AnnouncementUrl { get; set; }
    public string? FaviconUrl { get; set; }
    public string? LogoUrl { get; set; }
    public TenantStatusEnum Status { get; set; }
    public Guid AccountId { get; set; }
    public Account Account { get; set; } = null!;
    public ICollection<User> Users { get; set; } = [];
    public ICollection<Customer> Customers { get; set; } = [];
    public ICollection<Product> Products { get; set; } = [];
    public ICollection<Category> Categories { get; set; } = [];
}
